### DISCUSSION TITLE
Module 'immer' not found when using NextJS

### DISCUSSION THREAD

**Discussion by mauriciocrecencio** [pmndrs/zustand#1984] [Bug report] [answered]
I'm migrating a project from React to Next and when I tried to set up the store, this error shows up

```
./node_modules/zustand/esm/middleware/immer.mjs:1:0
Module not found: Can't resolve 'immer'
Did you mean './immer'?
Requests that should resolve in the current directory need to start with './'.
Requests that start with a name are treated as module requests and resolve within module directories (node_modules, /Users/mauriciocrecencio/freelas/frontend-conradotech).
If changing the source code is not an option there is also a resolve options called 'preferRelative' which tries to resolve these kind of requests in the current directory too.

https://nextjs.org/docs/messages/module-not-found


Import trace for requested module:
./src/store/productStore.ts
./src/components/Layout/index.tsx
./src/pages/index.tsx
```

To fix it I have to do `npm install immer`

Version:
zustand: ^4.4.1
next: 13.4.12

**Comment by dbritto-dev** [comment, marked as answer]
@mauriciocrecencio that's expected zustand immer middleware is an implementation on top of immer so you have to install `immer`

**Reply by mauriciocrecencio** [reply]
But `immer` should not be a dependecy of zustand and comes together when I install zustand?

**Reply by dbritto-dev** [reply]
> But `immer` should not be a dependecy of zustand and comes together when I install zustand?

No, shouldn't. If you want to use immer middleware you have to install immer package

**Reply by matovu-farid** [reply]
This should have been added to the docs. The docs make it seem like it is not needed

**Reply by dbritto-dev** [reply]
@matovu-farid btw, we're working on revamping docs
